### ISSUE TITLE
Output Location & Network Drive (help request)

### ISSUE THREAD

**Issue by bercaw** [lisamelton/video_transcoding#312]
When I stumbled upon the transcode-video tool, I was so inspired that I've been doing my best to learn CLI & bash. So far I've made good progress & I'm able to transcode mkvs on my desktop. I'm trying to take this to the next level and transcode files located on my NAS & output the new file to another folder on my NAS, but I keep getting an error that there's no such file or directory. I get this error even when I start the process by entering the bash command in the file explorer bar in either of the NAS folders. Apologies if this is the wrong place to post, I wasn't able to find a Q&A section for dummies. 

Btw, the transcode-video tool is magic - I'm incredibly thankful to have found it!

**Comment by damorrison** [comment]
Hi, I'll jump in with a few questions first that the collective hive mind would find useful to know how to help. 

- Firstly, are you using Linux or a Mac, or perhaps WSL on Windows?
- Have you mapped the drives so you can view them in a terminal/command window? i.e. can you do an `ls` on the volume where the files are?
- Can you post the exact error message you're getting, and perhaps the script you've written?

Some other thoughts:

- transcoding from a NAS to a NAS via another device will be a lot slower than transcoding them locally, i.e. copying the files to a local drive or perhaps an external USB3 drive first. 
- Don has written a new tool, called `other-transcode` which makes use of the hardware encoders available in most modern graphics cards, or Intel processors. You might want to check that out if you have a supported device as it allows for very fast transcoding, even with HEVC. You can find the new project here: https://github.com/donmelton/other_video_transcoding

**Comment by donmelton** [comment]
@bercaw Sorry I took so long to reply. But many thanks to @damorrison for stepping in while I was away sleeping!

Thank you for using my `transcode-video` tool. I'm glad you discovered it.

As to why you're getting that error when transcoding, I'm afraid I don't have quite enough information to diagnose it. But let me give a few suggestions since you appear to be new to the command line.

It's best to simply open a command line session in a separate window so it can be used interactively. Hopefully this is a Bash or PowerShell command line.

If so, use the `cd` command to change to the directory where you want your output to go. This is because `transcode-video` produces all its output in the current directory by default. So if you change the current directory (again, with the `cd` command) then you'll get that output exactly where you want.

That way you can specify which file you wish to transcode by using a full path to that file. On Windows that would be something like:

```
transcode-video "C:\path\to\Movie.mkv"
```

Obviously your full path to your original file would be very different. But hopefully you get the right idea here.

Does that help answer your question? Or have I misinterpreted you?

**Comment by Stooovie** [comment]
Can I change the output path to be always the original path (the path to original file), so I can have the outputs right next to the inputs? That would be my preferred way but I don't know how to do that. Thanks!

**Comment by donmelton** [comment]
@Stooovie Yes, but you should never do that. This is because the output path name would likely be the same as the input path name and I prevent you from overwriting your original.

Please, never write to the same directory as your original files. This is how horrible data loss accidents happen.
